cmd/osbuild-dnf-json-tests/main_test.go

```diff
@@ -47,25 +35,30 @@ func tearDownTemporaryRepository(dir string) error {
 	return os.RemoveAll(dir)
 }
 
-func TestFetchChecksum(quiet bool, dir string) {
+func TestFetchChecksum(t *testing.T) {
+	dir, err := setUpTemporaryRepository()
+	defer func(dir string) {
+		err := tearDownTemporaryRepository(dir)
+		if err != nil {
+			t.Errorf("Warning: failed to clean up temporary repository.")
+		}
+	}(dir)
+	if err != nil {
+		t.Errorf("Failed to set up temporary repository: %s", err)
+	}
+
 	repoCfg := rpmmd.RepoConfig{
 		Id:        "repo",
 		Name:      "repo",
 		BaseURL:   fmt.Sprintf("file://%s", dir),
 		IgnoreSSL: true,
 	}
-	if !quiet {
-		log.Println("Running TestFetchChecksum on:", dir)
-	}
 	rpmMetadata := rpmmd.NewRPMMD(path.Join(dir, "rpmmd"))
 	_, c, err := rpmMetadata.FetchMetadata([]rpmmd.RepoConfig{repoCfg}, "platform:f31")
 	if err != nil {
-		log.Panic("Failed to fetch checksum:", err)
+		t.Errorf("Failed to fetch checksum: %s", err)
```

Comment:
`t.Fatalf` might be worth considering here as well.

Reply:
Thanks again :)